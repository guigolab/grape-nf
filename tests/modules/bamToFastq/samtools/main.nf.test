nextflow_workflow {

    name "Test Workflow bamToFastq"
    script "modules/bamToFastq/samtools/main.nf"
    workflow "bamToFastq"

    test("Should convert BAM to FastQ (detecting protocol)") {

        when {
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments" ]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
