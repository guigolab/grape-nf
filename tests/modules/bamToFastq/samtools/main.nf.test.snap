{
    "Should convert BAM to FastQ (detecting protocol)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        [
                            "test3_1.fastq.gz:md5,acac9b626a3bba40b309c4fb880360f7",
                            "test3_2.fastq.gz:md5,877a915ea5eb84a315f41a1fa54bf5f1"
                        ],
                        "fastq",
                        [
                            "FqRd1",
                            "FqRd2"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T16:29:39.476232"
    }
}